### PR TITLE
PBRP: Bringup Deploy Scripts.

### PR DESCRIPTION
**Pull request by manjotsidhu**
The new deploy scripts are now more organized so that they can be scaled easily when required.
The code is not tested and hence **proper testing** is required.

Here the scenarios that needs to be tested via Circle CI Automated Builds -
- [x] Test Build
- [x] Official Build

Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .

### CODE REVIEW

**Comment by AndroiableDroid** [comment]
@manjotsidhu Beta should be released on beta channel not on group

File: pb_deploy.sh [hunk 1]
```diff
@@ -20,7 +20,8 @@
 #
 # Required Arguments: BUILD_TYPE(OFFICIAL/BETA/TEST)
```
Comment (manjotsidhu) [inline pb_deploy.sh]: Add the new arguments in the comments for documentation purpose

File: pb_deploy.sh [hunk 2]
```diff
@@ -160,12 +159,12 @@ function gh_deploy() {
 function tg_official_deploy() {
 	echo -e "${green}Deploying to Telegram!\n${nocol}"
 
-	FORMAT="PitchBlack Recovery for <b>$TARGET_DEVICE</b> (<code>${NAME}</code>)\n\n<b>Info</b>\n\nPitchBlack V${VERSION} ${DEPLOY_TYPE_NAME}\nBased on TWRP ${TWRP_V}\n<b>Build Date</b>: <code>${BUILD_DATE:0:4}/${BUILD_DATE:4:2}/${BUILD_DATE:6}</code>\n\n<b>Maintainer</b>: ${maintainer}\n"
+	FORMAT="PitchBlack Recovery for <b>$TARGET_DEVICE</b> (<code>${CODENAME}</code>)\n\n<b>Info</b>\n\nPitchBlack V${VERSION} ${DEPLOY_TYPE_NAME}\nBased on TWRP ${TWRP_V}\n<b>Build Date</b>: <code>${BUILD_DATE:0:4}/${BUILD_DATE:4:2}/${BUILD_DATE:6}</code>\n\n<b>Maintainer</b>: ${maintainer}\n"
```
Comment (manjotsidhu) [inline pb_deploy.sh]: Here can we change `TARGET_DEVICE` to actually get full device name stored in `pb_devices.json` ?

Comment (AndroiableDroid) [reply pb_deploy.sh]: hmmmm, lets see

**Comment by AndroiableDroid** [comment]
> The new deploy scripts are now more organized so that they can be scaled easily when required.
> The code is not tested and hence **proper testing** is required.
> 
> Here the scenarios that needs to be tested via Circle CI Automated Builds -
> 
> * [ ]  Test Build
> * [ ]  Official Build
> 
> Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .

flag changed to `PB_BETA := true`  also add Beta Build check

**Review by AndroiableDroid — COMMENTED** [review]
LGTM

**Comment by imranpopz** [comment]
> > The new deploy scripts are now more organized so that they can be scaled easily when required.
> > The code is not tested and hence **proper testing** is required.
> > Here the scenarios that needs to be tested via Circle CI Automated Builds -
> > 
> > * [ ]   Test Build
> > * [ ]   Official Build
> > 
> > Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .
> 
> flag changed to `PB_BETA := true` also add Beta Build check

I can see BETA_BUILD in 

> > The new deploy scripts are now more organized so that they can be scaled easily when required.
> > The code is not tested and hence **proper testing** is required.
> > Here the scenarios that needs to be tested via Circle CI Automated Builds -
> > 
> > * [ ]   Test Build
> > * [ ]   Official Build
> > 
> > Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .
> 
> flag changed to `PB_BETA := true` also add Beta Build check

In build.sh it is BETA_BUILD

**Comment by AndroiableDroid** [comment]
> > > The new deploy scripts are now more organized so that they can be scaled easily when required.
> > > The code is not tested and hence **proper testing** is required.
> > > Here the scenarios that needs to be tested via Circle CI Automated Builds -
> > > 
> > > * [ ]    Test Build
> > > * [ ]    Official Build
> > > 
> > > Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .
> > 
> > 
> > flag changed to `PB_BETA := true` also add Beta Build check
> 
> I can see BETA_BUILD in
> 
> > > The new deploy scripts are now more organized so that they can be scaled easily when required.
> > > The code is not tested and hence **proper testing** is required.
> > > Here the scenarios that needs to be tested via Circle CI Automated Builds -
> > > 
> > > * [ ]    Test Build
> > > * [ ]    Official Build
> > > 
> > > Further this now supports PBRP Official "Beta" Builds. A flag called `BETA_BUILD := true` is required to enable it. Beta Builds are released on PBRP's Official Testing Group i.e. https://t.me/pbrp_testers .
> > 
> > 
> > flag changed to `PB_BETA := true` also add Beta Build check
> 
> In build.sh it is BETA_BUILD

HMMM fine

**Review by manjotsidhu — COMMENTED** [review]
LGTM

**Review by AndroiableDroid — APPROVED** [review]

**Comment by AndroiableDroid** [comment]
@imranpopz test for test build

**Comment by manjotsidhu** [comment]
> @imranpopz test for test build

@raza231198

**Comment by manjotsidhu** [comment]
Test builds are working fine. We are good to go

**Comment by AndroiableDroid** [comment]
@manjotsidhu test now for SF and tell me so I will merge it

**Comment by manjotsidhu** [comment]
@AndroiableDroid Did you tested on local system ?

**Comment by AndroiableDroid** [comment]
> @AndroiableDroid Did you tested on local system ?

Yes

**Comment by manjotsidhu** [comment]
@AndroiableDroid Then I guess it's safe to merge ?

**Comment by AndroiableDroid** [comment]
I frequently tried uploading 7 times without any problems

**Comment by AndroiableDroid** [comment]
> @AndroiableDroid Then I guess it's safe to merge ?

Ya but you should try

**Comment by ahmedmoselhi** [comment]
Nice.. can I set flags now???

**Comment by manjotsidhu** [comment]
@ahmedmoselhi Yes

**Pull request merged**